src/main/webapp/assets/js/analytics.js

```diff
@@ -16,67 +16,54 @@
-              newSelect.appendChild(label);
               $('#termsSelect').on('change', function() {
+              	 $('.spinner').show();
                   var courseReportRequest = $.ajax({
                       url: 'report?action=getCoursesPublished&term=' + $(this).val() + '&termName=' + $('#termsSelect option:selected').text(),
                       type: 'GET',
                   });
                   courseReportRequest.done(function(response, status, xhr) {
-                      var filename = '';
-                      var disposition = xhr.getResponseHeader('Content-Disposition');
-                      if (disposition && disposition.indexOf('attachment') !== -1) {
-                          var filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
-                          var matches = filenameRegex.exec(disposition);
-                          if (matches !== null && matches[1]) filename = matches[1].replace(/['"]/g, '');
-                      }
-
-                      var type = xhr.getResponseHeader('Content-Type');
-                      var blob = new Blob([response], {
-                          type: type
-                      });
-
-                      if (typeof window.navigator.msSaveBlob !== 'undefined') {
-                          // IE workaround for "HTML7007: One or more blob URLs were revoked by closing the blob for which they were created. These URLs will no longer resolve as the data backing the URL has been freed."
-                          window.navigator.msSaveBlob(blob, filename);
-                      } else {
-                          var URL = window.URL || window.webkitURL;
-                          var downloadUrl = URL.createObjectURL(blob);
+                      var reportIntervId = setInterval(function() {
+                          reportTimer(response);
+                      }, reportPollInterv);
 
-                          if (filename) {
-                              // use HTML5 a[download] attribute to specify filename
-                              var a = document.createElement('a');
-                              // safari doesn't support this yet
-                              if (typeof a.download === 'undefined') {
-                                  window.location = downloadUrl;
-                              } else {
-                                  a.href = downloadUrl;
-                                  a.download = filename;
-                                  document.body.appendChild(a);
-                                  a.click();
+                      function reportTimer(response) {
+                      	 $('.spinner').show();
+                          var resultsMsg = $.ajax({
+                              url: 'report?action=polling&thrdId=' + response,
+                              type: 'get',
+                          }).done(function(response, status, xhr) {
+                              if (response !== "working") {
```

Comment:
Since the `WORKING` symbol is defined above and has the value "working", it should be used here, too.